src/batched/KokkosBatched_Vector_SIMD.hpp

```diff
@@ -135,6 +117,87 @@ namespace KokkosBatched {
 #if defined(KOKKOS_ENABLE_CUDA) && defined(__CUDA_ARCH__)
 namespace KokkosBatched {
 
+  template<>
+  class Vector<SIMD<float>,2> {
+  public:
+    using type = Vector<SIMD<float>,2>;
+    using value_type = float;
+    using mag_type = float;
+
+    enum : int { vector_length = 2 };
+    typedef float2 data_type;
```

Comment:
Could names like `float2` collide with existing types in the global namespace in e.g., CUDA?

Reply:
This is intended to override the cuda builtin type (float2, doule2, float4, double4).